ath79: Add support for TP-Link TL-WR842N-v2

robimarko opened the pull request:
This PR adds support for TP-Link TL-WR842N-v2 router which is supported by ar71xx to ath79.

This is a low cost model with following specs:

CPU: Atheros AR9341 SoC
RAM: 32 MB DDR1
Flash: 8 MB NOR SPI
Switch: Internal AR9341 5 port 10/100 Mbit
Ports:  5x 10/100 Mbit(1x WAN, 4x LAN)
USB: 1x USB2.0
WLAN: 2.4 GHZ AR9341

Installation:

Simply flash the factory image through stock firmware WEB UI.

Signed-off-by: Robert Marko <robimarko@gmail.com>

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -148,6 +148,11 @@ ath79_setup_interfaces()
 			"0@eth1" "1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1"
 		;;
+	tplink,tl-wr842-v2)
+		ucidef_set_interfaces_lan_wan "eth1.1" "eth0"
```
Hi, I think
ucidef_set_interface_wan "eth0"
is enough, since lan is set by the switch.

robimarko replied:
I will give it a go since I reused ar71xx configuration for network.

robimarko replied:
It works.
Will push it today

mkresin commented:
Hey Robert,

your PR has the following issues:

- e-mail address in commit message is malformed
- the board name looks wrong to me (tl-wr842-v2 vs. tl-wr842n-v2) and therefore adds a supported device for a board that doesn't exists with that name in ar71xx
- please drop the button postfix from the gpio keys. just "reset" and "wifi".

robimarko committed:
ath79: Add support for TP-Link TL-WR842N-v2

This PR adds support for TP-Link TL-WR842N-v2 router which is supported by ar71xx to ath79.

This is a low cost model with following specs:

CPU: Atheros AR9341 SoC
RAM: 32 MB DDR1
Flash: 8 MB NOR SPI
Switch: Internal AR9341 5 port 10/100 Mbit
Ports:  5x 10/100 Mbit(1x WAN, 4x LAN)
USB: 1x USB2.0
WLAN: 2.4 GHZ AR9341

Installation:

Simply flash the factory image through stock firmware WEB UI.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
@mkresin Your comments have been adressed.
Also redone the signed off line with git cli.

mkresin commented:
merged, thanks!
